### ISSUE TITLE
ValueError: min() arg is an empty sequence

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#671]
data: sct_testing_data/data/mt

error:
~~~
sct_process_segmentation -i left_dorsal_column.nii.gz -p csa -l 2:5 -t label/template/MNI-Poly-AMU_level.nii.gz
  OK: left_dorsal_column.nii.gz

Check parameters:
.. segmentation file:             left_dorsal_column.nii.gz

Create temporary folder...
mkdir tmp.151119161125_478478/

Copying input data to tmp folder and convert to nii...
sct_convert -i /Users/julien/data/temp/sct_testing_data/data/mt/left_dorsal_column.nii.gz -o tmp.151119161125_478478/segmentation.nii.gz

Change orientation to RPI...
sct_image -i segmentation.nii.gz -setorient RPI -o segmentation_RPI.nii.gz

Open segmentation volume...

Get data dimensions...
  40 x 40 x 5

Smooth centerline/segmentation...
.. Get center of mass of the centerline/segmentation...
.. Smoothing algo = hanning
.. Windows length = 50
Window size is too small. No smoothing was applied.
Window size is too small. No smoothing was applied.

Compute CSA...

Smooth CSA across slices...
.. Hanning window: 50 mm

Write text file...
z=0: 8.44098367816 mm^2
z=1: 7.57861092753 mm^2

Create volume of CSA values...
sct_image -i csa_volume_RPI.nii.gz -setorient RPI -o csa_volume.nii.gz

Generate output files...
  WARNING: File /Users/julien/data/temp/sct_testing_data/data/mt/csa_volume.nii.gz already exists. Deleting it...
  File created: /Users/julien/data/temp/sct_testing_data/data/mt/csa_volume.nii.gz

Find slices corresponding to vertebral levels...
WARNING: the bottom vertebral level you selected is lower to the lowest level available 
--> Selected the lowest vertebral level available: 3
Traceback (most recent call last):
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 689, in <module>
    main()
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 155, in main
    compute_csa(fname_segmentation, verbose, remove_temp_files, step, smoothing_param, figure_fit, param.file_csa_volume, slices, vert_lev, fname_vertebral_labeling, algo_fitting = param.algo_fitting, type_window= param.type_window, window_length=param.window_length)
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 499, in compute_csa
    slices, vert_levels_list, warning = get_slices_matching_with_vertebral_levels(Image(path_data+file_csa_volume).data, vert_levels, Image(fname_vertebral_labeling).data, 1)
  File "/Users/julien/code/spinalcordtoolbox/scripts/sct_extract_metric.py", line 479, in get_slices_matching_with_vertebral_levels
    distance_min_among_positive_value = min(abs(distance[distance > 0]))  # minimal distance among the negative
ValueError: min() arg is an empty sequence
~~~

**Comment by slevyrosetti** [comment]
I didn't succeed in reproducing the error on my station. Here is the output I get:
~~~
pass:~/test_batch_processing/sct_example_data/mt $ sct_maths -i label/atlas/WMtract__00.nii.gz -add label/atlas/WMtract__01.nii.gz -o left_dorsal_column.nii.gz

Done! To view results, type:
fslview left_dorsal_column.nii.gz &

pass:~/test_batch_processing/sct_example_data/mt $ sct_process_segmentation -i left_dorsal_column.nii.gz -p csa -l 2:5 -t label/template/MNI-Poly-AMU_level.nii.gz
  OK: left_dorsal_column.nii.gz

Check parameters:
.. segmentation file:             left_dorsal_column.nii.gz

Create temporary folder...
mkdir tmp.151119180533_78379/

Copying input data to tmp folder and convert to nii...
sct_convert -i /Users/slevy_local/test_batch_processing/sct_example_data/mt/left_dorsal_column.nii.gz -o tmp.151119180533_78379/segmentation.nii.gz

Change orientation to RPI...
sct_image -i segmentation.nii.gz -setorient RPI -o segmentation_RPI.nii.gz

Open segmentation volume...

Get data dimensions...
  55 x 58 x 5

Smooth centerline/segmentation...
.. Get center of mass of the centerline/segmentation...
.. Smoothing algo = hanning
.. Windows length = 50
Window size is too small. No smoothing was applied.
Window size is too small. No smoothing was applied.

Compute CSA...

Smooth CSA across slices...
.. Hanning window: 50 mm

Write text file...
z=0: 7.30950888274 mm^2
z=1: 9.14197613288 mm^2
z=2: 9.67630803509 mm^2
z=3: 10.3422758714 mm^2
z=4: 9.82633565674 mm^2

Create volume of CSA values...
sct_image -i csa_volume_RPI.nii.gz -setorient RPI -o csa_volume.nii.gz

Generate output files...
  File created: /Users/slevy_local/test_batch_processing/sct_example_data/mt/csa_volume.nii.gz

Find slices corresponding to vertebral levels...
  Check consistency of data size...
    OK!
  Find slices corresponding to vertebral levels...
    1:4
Average CSA across slices...
Mean CSA: 9.74672392403 +/- 0.42769686646 mm^2

Write results in /Users/slevy_local/test_batch_processing/sct_example_data/mt/csa_mean.txt...

Remove temporary files...
rm -rf tmp.151119180533_78379/

Done!
Output CSA volume: csa_volume.nii.gz
Output CSA file (averaged): csa_mean.txt
Output CSA file (all slices): csa.txt
~~~

**Comment by jcohenadad** [comment]
it's on testing data (not example data). I think this is related to the vertebral labeling: in case the user asks for levels that are not present. Try that under **sct_testing_data/data/mt**

~~~
sct_process_segmentation -i label/atlas/WMtract__02.nii.gz -p csa -l 1:30 -t label/template/MNI-Poly-AMU_level.nii.gz
  OK: label/atlas/WMtract__02.nii.gz

Check parameters:
.. segmentation file:             label/atlas/WMtract__02.nii.gz

Create temporary folder...
mkdir tmp.151120112758_964188/

Copying input data to tmp folder and convert to nii...
sct_convert -i /Users/julien/sct_testing_data/data/mt/label/atlas/WMtract__02.nii.gz -o tmp.151120112758_964188/segmentation.nii.gz

Change orientation to RPI...
sct_image -i segmentation.nii.gz -setorient RPI -o segmentation_RPI.nii.gz

Open segmentation volume...

Get data dimensions...
  40 x 40 x 5

Smooth centerline/segmentation...
.. Get center of mass of the centerline/segmentation...
.. Smoothing algo = hanning
.. Windows length = 50
Window size is too small. No smoothing was applied.
Window size is too small. No smoothing was applied.

Compute CSA...

Smooth CSA across slices...
.. Hanning window: 50 mm

Write text file...
z=0: 4.8083116903 mm^2
z=1: 5.61189012386 mm^2
z=2: 4.55261953487 mm^2

Create volume of CSA values...
sct_image -i csa_volume_RPI.nii.gz -setorient RPI -o csa_volume.nii.gz

Generate output files...
  WARNING: File /Users/julien/sct_testing_data/data/mt/label/atlas/csa_volume.nii.gz already exists. Deleting it...
  File created: /Users/julien/sct_testing_data/data/mt/label/atlas/csa_volume.nii.gz

Find slices corresponding to vertebral levels...
WARNING: the bottom vertebral level you selected is lower to the lowest level available 
--> Selected the lowest vertebral level available: 2
Traceback (most recent call last):
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 692, in <module>
    main()
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 155, in main
    compute_csa(fname_segmentation, verbose, remove_temp_files, step, smoothing_param, figure_fit, param.file_csa_volume, slices, vert_lev, fname_vertebral_labeling, algo_fitting = param.algo_fitting, type_window= param.type_window, window_length=param.window_length)
  File "/Users/julien/code/spinalcordtoolbox/bin/sct_process_segmentation", line 502, in compute_csa
    slices, vert_levels_list, warning = get_slices_matching_with_vertebral_levels(data_seg, vert_levels, im_vertebral_labeling.data, 1)
  File "/Users/julien/code/spinalcordtoolbox/scripts/sct_extract_metric.py", line 479, in get_slices_matching_with_vertebral_levels
    distance_min_among_positive_value = min(abs(distance[distance > 0]))  # minimal distance among the negative
ValueError: min() arg is an empty sequence
~~~

**Comment by slevyrosetti** [comment]
So, first, your initial command line (`-l 2:5`) works on my station with `sct_testing_data/data/mt`, here is the output:
~~~
pass:/Volumes/slevy-4/data/sct_testing_data/data/mt $ sct_process_segmentation -i label/atlas/WMtract__02.nii.gz -p csa -l 2:5 -t label/template/MNI-Poly-AMU_level.nii.gz
  OK: label/atlas/WMtract__02.nii.gz

Check parameters:
.. segmentation file:             label/atlas/WMtract__02.nii.gz

Create temporary folder...
mkdir tmp.151123115739_851646/

Copying input data to tmp folder and convert to nii...
sct_convert -i /Volumes/slevy-4/data/sct_testing_data/data/mt/label/atlas/WMtract__02.nii.gz -o tmp.151123115739_851646/segmentation.nii.gz

Change orientation to RPI...
sct_image -i segmentation.nii.gz -setorient RPI -o segmentation_RPI.nii.gz

Open segmentation volume...

Get data dimensions...
  40 x 40 x 5

Smooth centerline/segmentation...
.. Get center of mass of the centerline/segmentation...
.. Smoothing algo = hanning
.. Windows length = 50
Window size is too small. No smoothing was applied.
Window size is too small. No smoothing was applied.

Compute CSA...

Smooth CSA across slices...
.. Hanning window: 50 mm

Write text file...
z=0: 4.8083116903 mm^2
z=1: 5.61189012386 mm^2
z=2: 4.55261953487 mm^2

Create volume of CSA values...
sct_image -i csa_volume_RPI.nii.gz -setorient RPI -o csa_volume.nii.gz

Generate output files...
  WARNING: File /Volumes/slevy-4/data/sct_testing_data/data/mt/label/atlas/csa_volume.nii.gz already exists. Deleting it...
  File created: /Volumes/slevy-4/data/sct_testing_data/data/mt/label/atlas/csa_volume.nii.gz

Find slices corresponding to vertebral levels...
  Check consistency of data size...
    OK!
  Find slices corresponding to vertebral levels...
    1:4
Average CSA across slices...
Mean CSA: 4.99094044968 +/- 0.451315479451 mm^2

Write results in /Volumes/slevy-4/data/sct_testing_data/data/mt/label/atlas/csa_mean.txt...

Remove temporary files...
rm -rf tmp.151123115739_851646/
rm: tmp.151123115739_851646/: Directory not empty

Done!
Output CSA volume: csa_volume.nii.gz
Output CSA file (averaged): csa_mean.txt
Output CSA file (all slices): csa.txt
~~~
Just becareful: I'm not sure if it is on purpose or not but the labels are only registered to the levels 2 to 5 whereas the MT images were acquired from levels 2 to 6...

Concerning your second command line (`-l 1:30`), I get the same error on my station. Indeed, it seems to be due to the fact that the user asked for unavailable levels towards the inferior direction. In the superior direction, the script takes the lowest level available and display a warning message.
I will modify the script to take it into account.

**Comment by slevyrosetti** [comment]
Fixed in commit #0307345

**Comment by jcohenadad** [comment]
yes, i indeed meant the second command line-- your fix is good.
thanks
